.github/workflows/nodejs-ci.yml

```diff
@@ -0,0 +1,33 @@
+name: Intacct JS SDK CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+
+    name: Node.js v${{ matrix.node-version }}
+    
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v3
+      
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
```

Comment:
since you are using the same cache key `npm`  for all versions, I am afraid all matrix versions will be cached against the same key. Can you concatenate `matrix-node-version` to `npm` so each version will have its own cache key?